transport-native-unix-common/src/main/c/netty_unix_socket.c

```diff
@@ -390,6 +394,36 @@ static jint netty_unix_socket_finishConnect(JNIEnv* env, jclass clazz, jint fd)
     return -optval;
 }
 
+static jint netty_unix_socket_disconnect(JNIEnv* env, jclass clazz, jint fd) {
+    struct sockaddr_storage addr;
+    int len;
+
+    memset(&addr, 0, sizeof(addr));
+
+    // You can disconnect connection-less sockets by using AF_UNSPEC.
+    // See man 2 connect.
+    if (socketType == AF_INET6) {
+        struct sockaddr_in6* ip6addr = (struct sockaddr_in6*) &addr;
+        ip6addr->sin6_family = AF_UNSPEC;
+        len = sizeof(struct sockaddr_in6);
+    } else {
+        struct sockaddr_in* ipaddr = (struct sockaddr_in*) &addr;
+        ipaddr->sin_family = AF_INET;
+        len = sizeof(struct sockaddr_in);
+    }
+
+    int res;
+    int err;
+    do {
+        res = connect(fd, (struct sockaddr*) &addr, len);
+    } while (res == -1 && ((err = errno) == EINTR));
```

Comment:
Do we need to account for `EAFNOSUPPORT`? Here is `man 2 connect` on macOS:

> Datagram sockets may dissolve the association by calling
     disconnectx(2), or by connecting to an invalid address, such as a null address or an address with the address family set to AF_UNSPEC (the
     error EAFNOSUPPORT will be harmlessly returned).